themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/signingin-page/SigningInPage.tsx

```diff
@@ -0,0 +1,183 @@
+    
+    public constructor(props: SigningInPageProps) {
+        super(props);
+        this.state = {
+            twoFactorEnabled: true,
+            twoFactorEnabledText: Msg.localize('twoFactorEnabled'),
+            isTotpConfigured: features.isTotpConfigured,
+        }
+        this.setLastPwdUpdate();
+    }
+
+    private setLastPwdUpdate(): void {
+        AccountServiceClient.Instance.doGet("/credentials/password")
+            .then((response: AxiosResponse<PasswordDetails>) => {
+                if (response.data.lastUpdate) {
+                    const lastUpdate: number = response.data.lastUpdate;
+                    this.setState({lastPasswordUpdate: lastUpdate});
+                }
+            });
+    }
+    
+    private handleTwoFactorSwitch = () => {
+        if (this.state.twoFactorEnabled) {
+            this.setState({twoFactorEnabled: false, twoFactorEnabledText: Msg.localize('twoFactorDisabled')})
+        } else {
+            this.setState({twoFactorEnabled: true, twoFactorEnabledText:  Msg.localize('twoFactorEnabled')})
+        }
+    }
+
+    private handleRemoveTOTP = () => {
+      AccountServiceClient.Instance.doDelete("/totp/remove")
+        .then(() => {
+            this.setState({isTotpConfigured: false});
+            ContentAlert.success('successTotpRemovedMessage');
+        });
+    }
+
+    public render(): React.ReactNode {
+        let lastPwdUpdate: string = Msg.localize('unknown');
+        if (this.state.lastPasswordUpdate) {
+            lastPwdUpdate = moment(this.state.lastPasswordUpdate).format('LLL');
+        }
+
+        return (
+            <ContentPage title="signingIn" 
+                     introMessage="signingInSubMessage">
+                <Stack gutter='md'>
+                    <StackItem isFilled>
+                        <Title headingLevel={TitleLevel.h2} size='2xl'>
+                        <strong><Msg msgKey='password'/></strong>
+                        </Title>
+                        <DataList aria-label='foo'>
+                            <DataListItemRow>
+                                <DataListItemCells
+                                    dataListCells={[
+                                        <DataListCell key='password'><Msg msgKey='password'/></DataListCell>,
+                                        <DataListCell key='lastPwdUpdate'><strong><Msg msgKey='lastUpdate'/>: </strong>{lastPwdUpdate}</DataListCell>,
+                                        <DataListCell key='spacer'/>
+                                    ]}/>
+                                <DataListAction aria-labelledby='foo' aria-label='foo action' id='setPasswordAction'>
+                                    <Button variant='primary'onClick={()=> this.updatePassword.execute()}><Msg msgKey='update'/></Button>
+                                </DataListAction>
+                            </DataListItemRow>
```

Comment:
We should somehow reflect when no password is set. Currently it only displays "Last Update: Unknown" which is not enough. We need to clearly indicate that no password is set.

Reply:
In the case where passwords are not being used, we should not display the Password section at all.

If password is optional and has never been set, then I agree it would be better to indicate that.

I'll leave it up to Bruno whether he wants to try to implement something along those lines or write a JIRA to take care of it after tech preview.  I'd prefer not to spend lots of time figuring out what to do before tech preview.